tests/integration_tests/instances.py

```diff
@@ -83,36 +92,54 @@ def write_to_file(self, remote_path, contents: str):
             os.unlink(tmp_file.name)
 
     def snapshot(self):
-        return self.cloud.snapshot(self.instance)
-
-    def _install_new_cloud_init(self, remote_script):
-        self.execute(remote_script)
+        image_id = self.cloud.snapshot(self.instance)
+        log.info('Created new image: %s', image_id)
+        return image_id
+
+    def install_new_cloud_init(
+        self,
+        source: CloudInitSource,
+        take_snapshot=True
+    ):
+        if source == CloudInitSource.DEB_PACKAGE:
+            self.install_deb()
+        elif source == CloudInitSource.PPA:
+            self.install_ppa()
+        elif source == CloudInitSource.PROPOSED:
+            self.install_proposed_image()
```

Comment:
Worth logging a warning in an `else` here?

Reply:
I gave you an exception instead  :slightly_smiling_face: